src/tests/suite/daos_nvme_recovery.c

```diff
@@ -119,9 +119,183 @@ nvme_recov_1(void **state)
+	 * different state of NVMe drives. Skip the test if log_mask is not
+	 * set to DEBUG.
+	 */
+	D_ALLOC(server_config_file, 512);
+	D_ALLOC(log_file, 1024);
+	rc = get_server_config(devices[rank_pos].host,
+			       server_config_file);
+	assert_int_equal(rc, 0);
+	print_message("server_config_file = %s\n", server_config_file);
+
+	get_server_log_file(devices[rank_pos].host,
+			    server_config_file, log_file);
+	rc = verify_server_log_mask(devices[rank_pos].host,
+				    server_config_file, "DEBUG");
+	if (rc) {
+		print_message("Log Mask != DEBUG in %s.\n",
+			      server_config_file);
+		skip();
+	}
+
+	print_message("LOG FILE = %s\n", log_file);
+
+	/** Prepare records **/
+	if (arg->pool.pool_info.pi_nnodes < 2)
+		obj_class = DAOS_OC_R1S_SPEC_RANK;
+	else
+		obj_class = DAOS_OC_R2S_SPEC_RANK;
+
+	oid = dts_oid_gen(obj_class, 0, arg->myrank);
+	oid = dts_oid_set_rank(oid, 0);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+	memset(data_buf, 'a', 100);
+
+	/** Insert record **/
+	print_message("Insert single record with 100 extents\n");
+	insert_single_with_rxnr("dkey", "akey", idx, data_buf,
+				1, 100, DAOS_TX_NONE, &req);
+
+	/**
+	*Set single device for rank0 to faulty.
+	*/
+	print_message("NVMe with UUID=%s on host=%s\" set to Faulty\n",
+		      DP_UUID(devices[rank_pos].device_id),
+		devices[rank_pos].host);
+	rc = dmg_storage_set_nvme_fault(dmg_config_file,
+					devices[rank_pos].host,
+		devices[rank_pos].device_id, 1);
+	assert_int_equal(rc, 0);
+	sleep(60);
+
+	/**
+	* Verify Rank0 device state change from NORMAL to FAULTY.
+	* Verify "FAULTY -> TEARDOWN" and "TEARDOWN -> OUT" device states found
+	* in server log.
+	*/
+	rc = dmg_storage_device_list(dmg_config_file, NULL, devices);
+	assert_int_equal(rc, 0);
+	assert_string_equal(devices[rank_pos].state, "\"FAULTY\"");
+
+	rc = verify_state_in_log(devices[rank_pos].host, log_file,
+				 "NORMAL -> FAULTY");
+	if (rc != 0) {
+		print_message("NORMAL -> FAULTY not found in log %s\n",
+			      log_file);
+		assert_int_equal(rc, 0);
+	}
+
+	rc = verify_state_in_log(devices[rank_pos].host, log_file,
+				 "FAULTY -> TEARDOWN");
+	if (rc != 0) {
+		print_message("FAULTY -> TEARDOWN not found in %s\n",
+			      log_file);
+		assert_int_equal(rc, 0);
+	}
+
+	rc = verify_state_in_log(devices[rank_pos].host, log_file,
+				 "TEARDOWN -> OUT");
+	if (rc != 0) {
+		print_message("TEARDOWN -> OUT not found in log %s\n",
+			      log_file);
+		assert_int_equal(rc, 0);
+	}
+
+	/*
+	 *  Insert fresh record because total rank < 1 and disk fault will
+	 *  lost the data.
+	 */
+	if (read_initial_data != true) {
+		idx = 1;
+		print_message("Insert fresh record with 100 extents\n");
+		insert_single_with_rxnr("dkey", "akey", idx, data_buf,
+					1, 100, DAOS_TX_NONE, &req);
+	}
+
+	/** Lookup all the records and verify the content **/
+	print_message("Lookup and Verify records:\n");
+	lookup_single_with_rxnr("dkey", "akey", idx, fetch_buf,
+				1, 100, DAOS_TX_NONE, &req);
```

Comment:
I don't really think lookup and verify of all records is required for this specific test. I suggest simplifying the test to just verifying the states and no the data.
Also, when running this with 1 server and 2 disks, the verification fails so something is wrong there anyways
Found state NORMAL -> FAULTY in Log file  /tmp/smvanda/daos_server.log     # map to D_LOG_FILE=/tmp/server.log
Found state FAULTY -> TEARDOWN in Log file  /tmp/smvanda/daos_server.log     # map to D_LOG_FILE=/tmp/server.log
Found state TEARDOWN -> OUT in Log file  /tmp/smvanda/daos_server.log     # map to D_LOG_FILE=/tmp/server.log
Insert fresh record with 100 extents
Lookup and Verify records:
[  ERROR   ] --- 0xfffffffffffffc13 != 0
[   LINE   ] --- src/tests/suite/daos_obj.c:453: error: Failure!

Reply:
Will remove the data verification.